test_project/store/src/main/java/test/sergeyvalyushko/store/handlers/DisplayCatalog.java

```diff
@@ -0,0 +1,34 @@
+package test.sergeyvalyushko.store.handlers;
+
+import org.xml.sax.SAXException;
+import test.sergeyvalyushko.store.Catalog;
+import test.sergeyvalyushko.store.Category;
+import test.sergeyvalyushko.store.Product;
+
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.IOException;
+
+public class DisplayCatalog extends AbstractHandler {
+    public DisplayCatalog(AbstractHandler nextHandler) {
+        super(nextHandler);
+    }
+
+    @Override
+    public void handleRequest(String input, Catalog catalog) throws ParserConfigurationException, SAXException, IOException {
+        if (input.equals("init")) {
+            for (Category category : catalog.getCategories()) {
+                if (category.getProductList().isEmpty()) {
```

Comment:
it is very bad solution, cause you populate the catalog in the wrong place, it should not be done in display command.